distrobox-export

```diff
@@ -228,8 +235,17 @@ if [ -z "${container_name}" ]; then
 	container_name=$(uname -n | cut -d'.' -f1)
 fi
 
+#
+if [ "${is_login}" -ne 0 ] && [ "${is_sudo}" -ne 0 ]; then
+	start_shell="sudo -i"
+elif [ "${is_login}" -ne 0 ]; then
+	start_shell="sh -l -c"
```

Comment:
This does not work for commands taking arguments since `-c` takes a string, and not arbitrary parameters. Also shouldn't it use the user shell rather than `sh`

Reply:
Thanks @fredizzimo I've pushed a fix for it :)

Reply:
Looking a bit further, there is another issue that I don't quite understand. If you run `bash gcc`, it will abort with `cannot execute binary file`, whereas `bash -c "gcc"` successfully runs. It seems to me that having a structure like this:
```
${SHELL} -l -c "exportedCommand $@"
```
is the most general solution.

Reply:
In order to work around the issues I had here https://github.com/89luca89/distrobox/issues/656 I made changes to always run the login shell, no matter what. In `distrobox-enter`  

```diff
        result_command="${result_command} ${container_name}"
 
        if [ -n "${container_command}" ]; then
-               result_command="${result_command} ${container_command}"
+                quoted_container_command="\\\"${container_command}\\\""
+               result_command="${result_command} sh -c \"\\\$(getent passwd ${USER} | cut -f 7 -d :) -lc ${quoted_con
tainer_command}\""
        else
                # if no command was specified, let's execute a command that will find
                # and run the default shell for the user
```
So that's basically what's @Scafir is suggesting to do here, allthough the quoting needed special handling there. I'm not sure if it's needed here as well.

By always running the login shell I can guarantee that the rc file is always run and I can unset and set the environment variables I want. This variant has been working well for me, but I'm not sure if that's the best way to solve the issues, so I have not sent any PR .But anyway I think a similar approach could be taken here as well.